Composer/packages/client/src/components/CreationFlow/DefineConversation/index.tsx

```diff
@@ -146,22 +127,24 @@ const DefineConversation: React.FC<DefineConversationProps> = (props) => {
       } else {
         updatedFormData.name = getDefaultName();
       }
-      setFormData(updatedFormData);
+      updateForm(updatedFormData);
     }
   }, [templateId]);
 
-  const handleSubmit = (e) => {
-    e.preventDefault();
-    const errors = validateForm(formData);
-    if (Object.keys(errors).length) {
-      setFormDataErrors(errors);
-      return;
-    }
+  const handleSubmit = useCallback(
+    (e) => {
+      e.preventDefault();
+      if (hasErrors) {
+        return;
+      }
+
+      onSubmit({
```

Comment:
We are memoizing this function if formData or hasErrors changes. But, it still is going to call the onSubmit even if there is no change to formData here though right. should we check for changes in formData or do u think its trivial change and it can continue to submit even if no changes were present

Reply:
I think in this case it's fine. We just validate that the `formData` is valid before submitting. So when `hasErrors` changes, we need a new function that reads that value.